app/gen-server/lib/homedb/HomeDBManager.ts

```diff
@@ -692,6 +692,36 @@ export class HomeDBManager extends EventEmitter {
           ws.owner = o.owner;
           // Include the org's domain so that the UI can build doc URLs that include the org.
           ws.orgDomain = o.domain;
+
+          // Include shareType based on users permissions to set icon color
+          const {maxInheritedRole, users } =  this.unwrapQueryResult(
+            await this.getWorkspaceAccess({userId: scope.userId}, ws.id, true)
+          );
+          const permissionDataUsers = users.filter((user) => {
+            // effectiveRole - Gives the effective role of the member on the resource, taking everything into account.
+            return !!roles.getStrongestRole(user.access,
+              roles.getWeakestRole(<roles.Role>user.parentAccess, <roles.Role>maxInheritedRole));
```

Comment:
There are much chances that I misunderstand something. Please don't hesitate to tell me if I am wrong :).

I expect this expression `roles.getStrongestRole(user.access, roles.getWeakestRole(<roles.Role>user.parentAccess, <roles.Role>maxInheritedRole));` to return the effective role as described in the comment above. 


So we filter anyone having at least a direct access or an inherited access.

It would thus be equivalent to:
```ts
const hasDirectAccess = Boolean(user.access);
const hasInheritedAccess = Boolean(roles.getWeakestRole(<roles.Role>user.parentAccess, <roles.Role>maxInheritedRole)));
return hasDirectAccess || hasInheritedAccess;
```

If so, if you prefer your current implementation, could you add a comment to clarify the that?